shiny/express/_mock_session.py

```diff
@@ -24,11 +30,24 @@ class ExpressMockSession:
     def __init__(self, ns: ResolvedId = Root):
         self.ns = ns
         self.input = Inputs({})
-        self.output = Outputs(cast(Session, self), self.ns, outputs={})
+        self.output = Outputs(self, self.ns, outputs={})
+
+        # Set some of these values to None just to satisfy the abstract base class to
+        # make this code run -- these things should not be used at run time, so None
+        # will work as a placeholder. But we also need to tell pyright to ignore that
+        # the Nones don't match the type declared in the Session abstract base class.
+        self.app = None  # pyright: ignore
+        self.id = None  # pyright: ignore
+
+        self._outbound_message_queues = None  # pyright: ignore
+        self._downloads = None  # pyright: ignore
 
         # Application-level (not session-level) options that may be set via app_opts().
         self.app_opts: AppOpts = {}
 
+    def is_real_session(self) -> Literal[False]:
```

Comment:
Regarding the name of this method, eventually we'll probably have a MockSession that's used for testing, including testing server code (so would "implement" a lot more than ExpressMockSession, I'd imagine). If that happens, `is_real_session` becomes a bit muddy.

`is_server_session`? `is_user_session`?

Naming things is hard...